hera_cal/io.py

```diff
@@ -388,23 +394,26 @@ def read(self, bls=None, polarizations=None, times=None,
         partials = ['bls', 'polarizations', 'times', 'frequencies', 'freq_chans']
         self.last_read_kwargs = {p: locs[p] for p in partials}
 
-        # load data
-        if self.filetype == 'uvh5':
-            self.read_uvh5(self.filepaths, bls=bls, polarizations=polarizations, times=times,
-                           frequencies=frequencies, freq_chans=freq_chans, read_data=read_data)
-        else:
-            if not read_data:
-                raise NotImplementedError('reading only metadata is not implemented for ' + self.filetype)
-            if self.filetype == 'miriad':
-                self.read_miriad(self.filepaths, bls=bls, polarizations=polarizations)
-                if any([times is not None, frequencies is not None, freq_chans is not None]):
-                    warnings.warn('miriad does not support partial loading for times and frequencies. '
-                                  'Loading the file first and then performing select.')
-                    self.select(times=times, frequencies=frequencies, freq_chans=freq_chans)
-            elif self.filetype == 'uvfits':
-                self.read_uvfits(self.filepaths, bls=bls, polarizations=polarizations,
-                                 times=times, frequencies=frequencies, freq_chans=freq_chans)
-                self.unphase_to_drift()
+        # if filepaths is None, this was converted to HERAData
+        # from a different pre-loaded object with no history of filepath
+        if self.filepaths is not None:
+            # load data
+            if self.filetype == 'uvh5':
+                self.read_uvh5(self.filepaths, bls=bls, polarizations=polarizations, times=times,
+                               frequencies=frequencies, freq_chans=freq_chans, read_data=read_data)
+            else:
+                if not read_data:
+                    raise NotImplementedError('reading only metadata is not implemented for ' + self.filetype)
+                if self.filetype == 'miriad':
+                    self.read_miriad(self.filepaths, bls=bls, polarizations=polarizations)
+                    if any([times is not None, frequencies is not None, freq_chans is not None]):
+                        warnings.warn('miriad does not support partial loading for times and frequencies. '
+                                      'Loading the file first and then performing select.')
+                        self.select(times=times, frequencies=frequencies, freq_chans=freq_chans)
+                elif self.filetype == 'uvfits':
+                    self.read_uvfits(self.filepaths, bls=bls, polarizations=polarizations,
+                                     times=times, frequencies=frequencies, freq_chans=freq_chans)
+                    self.unphase_to_drift()
```

Comment:
Don't forget to add errors if you try to do a select and there's no self.filepaths

Reply:
Ah, ok. Good point.